packages/data-broker/src/services/dhis/DhisService.js

```diff
@@ -226,27 +197,81 @@ export class DhisService extends Service {
     return buildAnalyticsFromEvents(translatedEvents, dataElements);
   };
 
-  pullAnalyticsFromEventsForApi = async (api, dataSources, options) => {
-    const { programCodes = [], period, startDate, endDate } = options;
+  translateDhisEventAnalytics = async (dhisAnalytics, dataSources) => {
```

Comment:
Do all these little, specific functions it easier to understand or less so?

Reply:
I love them! Size does matter in function length: all other qualitative aspects being equal, the smaller the better!

A few resources on the subject:
* [Martin Fowler - FunctionLength](https://martinfowler.com/bliki/FunctionLength.html)
* [Rule of 30](https://dzone.com/articles/rule-30-%E2%80%93-when-method-class-or)

Reply:
Awesome! I finally found time to read those sources and they seem pretty clear! I'm glad to hear that (although perhaps they should be even shorter (`mergeAnalyticResults`, `mergeAnalyticsMetadata`)